invenio_access/permissions.py

```diff
@@ -76,13 +90,20 @@ def _load_permissions(self):
                 ).all()
 
                 for actionrole in actionsroles:
-                    result[actionrole.exclude].add(
+                    action[actionrole.exclude].add(
                         actionrole.need
                     )
+
+                result[True] |= action[True]
+                result[False] |= action[False]
+
+                if cache:
+                    cache.set(cache_key, action)
+
             else:
                 result[self.NEEDS].add(explicit_need)
 
-        self._permissions = result
```

Comment:
@jirikuncar But if you use a local variable storing the permissions after ```_load_permission``` execution, you will have to clear it after an ```ActionUsers``` or ```ActionRoles``` update, and this is not trivial.